File: 2838/2838.cs
public class Solution {
    public int MaximumCoinsHeroesCanCollect(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(MaximumCoinsHeroesCanCollect(root.left), MaximumCoinsHeroesCanCollect(root.right));
    }
}

File: 2838/2838.py
class Solution:
    def maximum_coins_heroes_can_collect(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        return 1 + max(self.maximum_coins_heroes_can_collect(root.left), self.maximum_coins_heroes_can_collect(root.right))